### PR TITLE
refactor: extract icon scale adjustment logic into utility function

### PR DESCRIPTION
**Pull request by alhridoy**
**Related Issue:** #7765 

## Summary
This pull request extracts the logic for adjusting internal icon scales into a separate utility function. This function is now used across all components, making the code DRYer and easier to maintain.

## Changes
- Created a new utility function `adjustIconScale` in `iconScaleAdjuster.ts`.
- Replaced inline logic for adjusting icon scales with calls to `adjustIconScale` in all relevant components.

## Testing
- All unit tests pass.
- Manual testing in the browser confirms that icons still display correctly at all scales.

This change stems from issue #7755 and is expected to make the codebase easier to maintain by reducing repetition of this logic.

### CODE REVIEW

File: packages/calcite-components/src/components/accordion-item/accordion-item.tsx [hunk 1]
```diff
@@ -21,6 +21,9 @@ import {
   toAriaBoolean,
 } from "../../utils/dom";
 import { CSS_UTILITY } from "../../utils/resources";
+import { adjustIconScale } from "../../utils/iconScaleAdjuster";
+
+// ...
```
Comment (jcfranco) [inline packages/calcite-components/src/components/accordion-item/accordion-item.tsx:26]: Can you remove this comment?

File: packages/calcite-components/src/components/action/action.tsx [hunk 2]
```diff
@@ -248,13 +249,13 @@ export class Action
 
   renderIconContainer(): VNode {
     const { loading, icon, scale, el, iconFlipRtl } = this;
-    const iconScale = scale === "l" ? "m" : "s";
+    //const iconScale = scale === "l" ? "m" : "s";
```
Comment (jcfranco) [inline packages/calcite-components/src/components/action/action.tsx:252]: Can you remove this commented out line?

File: packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { adjustIconScale } from "./iconScaleAdjuster";
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts:1]: Could you rename the util module to `component.ts` and the exported function to the following?

```ts
getIconScale(componentScale: Scale): Extract<Scale, "s" | "m">;
```

**Review by jcfranco — CHANGES_REQUESTED** [review]
Looking great! Just a few tweaks and we can get this moving forward. 🚀

**Comment by jcfranco** [comment]
I've gone ahead and updated the PR title to follow conventional commits (per [our contributing guidelines](https://github.com/Esri/calcite-design-system/blob/main/CONTRIBUTING.md#commit-message-format)).

**Comment by alhridoy** [comment]
Hi @jcfranco Please check my new commits and let me know if everything is fine now. Thanks for your guidance.

File: packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { getIconScale } from "./iconScaleAdjuster";
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts:1]: This file is no longer needed.

**Review by jcfranco — COMMENTED** [review]
Other than removing `iconScaleAdjuster.spec.ts` and resolving merge conflicts, this LGTM. 

Once the last changes are made, we can take it from there to get screenshot tests to run. Thanks again! 🚀

File: packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts (hunk 4 above)
Comment (alhridoy) [reply packages/calcite-components/src/utils/iconScaleAdjuster.spec.ts]: Hi @jcfranco removed the file, kindly check. Thanks

**Comment by alhridoy** [comment]
Hi @jcfranco i am just checking if  everything is ok on the latest changes. If I need to modify anything please let me know. Thanks

**Comment by jcfranco** [comment]
@alhridoy The PR still shows merge conflicts, so once these are resolved, we can take it from there. Thanks again for helping out with this!

**Comment by jcfranco** [comment]
I went ahead and adopted this PR (#7973) to get both E2E and screenshot tests running. Once checks are ✅, we should be good to install for the October release.

**Comment by alhridoy** [comment]
Hi @jcfranco Thanks. If test pass you are going to merge the PR, right?

**Comment by jcfranco** [comment]
@alhridoy The adopted one (containing your original commits), yes.

**Comment by jcfranco** [comment]
Merged via #7973, closing!

Many thanks, @alhridoy!

**Comment by alhridoy** [comment]
Hi @jcfranco Great! You welcome. Thanks a lot for guidances.

**Pull request closed without merging**